### ISSUE TITLE
Setting shell menu item ("ShellContent") IsVisible to false breaks unrelated page menu item or routing

### ISSUE THREAD

**Issue by janseris** [dotnet/maui#14469]
### Description

When
![image](https://user-images.githubusercontent.com/64279914/230718269-0b85af61-077e-4ec9-a946-bab52377aa1c.png)

Then
`System.InvalidOperationException: 'Content not found for active Microsoft.Maui.Controls.ShellItem. Title: Monitoring. Route: IMPL_MonitoringPage.'` (Android)

(On Windows I get just unhelpful NullReferenceException)

### Steps to Reproduce

1. Change `IsVisible="True"` to `IsVisible="False"` in first shell menu item code (`line 20` in `AppShell.xaml`)
2. Run application

### Link to public reproduction project repository

https://github.com/janseris/MAUIDynamicMenuItemsBugReports

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

Android, Windows

### Affected platform versions

Android 12, windows10.0.19041.0

### Did you find any workaround?

No

**Comment by janseris** [comment]
The same happens when the LoginPage shell menu item is removed.

**Comment by janseris** [comment]
The same does **_not_** happen when `FlyoutItemIsVisible` is set to `False` instead of `IsVisible`

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.7.0 Preview 2.0. Repro on Windows 11 and 13.0-API33 with below Project:
[DynamicMenu.zip](https://github.com/dotnet/maui/files/11877064/DynamicMenu.zip)

**Comment by Perryyang1979** [comment]
I have the same problem , I try to implement the function that, some menu Item will be invisible before the login .
So I binding the IsVisible to a property which will cause the exception as this bug mentioned.

`<ShellItem  FlyoutItemIsVisible="false">
    <ShellContent ContentTemplate="{DataTemplate view:LoginPage}" Route="LoginPage"/>
</ShellItem>


<FlyoutItem Title="Home" FlyoutDisplayOptions="AsMultipleItems">
    <ShellContent Icon="home.png" Title="Home" IsVisible="{Binding HomeItemEnable}"
    ContentTemplate="{DataTemplate view:AddTransactionPage}"
    Route="AddTransactionPage" />

    <ShellContent Icon="report.png" Title="Report"  IsVisible="{Binding HomeItemEnable}"
    ContentTemplate="{DataTemplate view:AdvancedReportPage}"
    Route="AdvancedReportPage" />

    <ShellContent Icon="me.png" Title="Me" IsVisible="{Binding HomeItemEnable}"
           ContentTemplate="{DataTemplate view:UserInfoPage}"
       Route="UserInfoPage"  />

    <ShellContent Icon="setting.png" Title="Setting"
    ContentTemplate="{DataTemplate view:SettingPage}"
    Route="SettingPage" />
</FlyoutItem>`

**Comment by janseris** [comment]
> I have the same problem , I try to implement the function that, some menu Item will be invisible before the login . So I binding the IsVisible to a property which will cause the exception as this bug mentioned.
> 
> `
> 
> ```
> <ShellContent Icon="report.png" Title="Report"  IsVisible="{Binding HomeItemEnable}"
> ContentTemplate="{DataTemplate view:AdvancedReportPage}"
> Route="AdvancedReportPage" />
> 
> <ShellContent Icon="me.png" Title="Me" IsVisible="{Binding HomeItemEnable}"
>        ContentTemplate="{DataTemplate view:UserInfoPage}"
>    Route="UserInfoPage"  />
> 
> <ShellContent Icon="setting.png" Title="Setting"
> ContentTemplate="{DataTemplate view:SettingPage}"
> Route="SettingPage" />
> ```
> 
> `

Try using FlyoutItemIsVisible as I did. It is some kind of redundant property which I think worked

**Comment by Perryyang1979** [comment]
``> property

However ,the property FlyoutItemIsVisible is not a bindable , not sure how we can make it works.

I try to use IsVisible  property for FlyoutItem instead of FlyoutItemIsVisible ,also instead of ShellContent 
`
    <ShellItem  FlyoutItemIsVisible="false">
        <ShellContent ContentTemplate="{DataTemplate view:LoginPage}" Route="LoginPage"/>
    </ShellItem>


    <FlyoutItem Title="Home" FlyoutDisplayOptions="AsMultipleItems" IsVisible="{Binding HomeItemEnable}">
        <ShellContent Icon="home.png" Title="Home" 
        ContentTemplate="{DataTemplate view:AddTransactionPage}"
        Route="AddTransactionPage" />

        <ShellContent Icon="report.png" Title="Report" 
        ContentTemplate="{DataTemplate view:AdvancedReportPage}"
        Route="AdvancedReportPage" />

        <ShellContent Icon="me.png" Title="Me" 
               ContentTemplate="{DataTemplate view:UserInfoPage}"
           Route="UserInfoPage"  />
      
    </FlyoutItem>

    <FlyoutItem Icon="setting.png" Title="Setting" FlyoutDisplayOptions="AsSingleItem" >
        <ShellContent 
ContentTemplate="{DataTemplate view:SettingPage}" Icon="setting.png" Title="Setting"
Route="SettingPage" />
    </FlyoutItem>`

I try to separate 4 shell content to 2 FlyoutItem  ,one is binding the property Isvisible to make sure it will be changed danamic .
and another FlyoutItem   is static ,always show this item .
However the problem is that the routing page go to login page(expectation: go t setting page ) when I click setting menu if 4 of shell content are  visible.

**Comment by Shalini-Ashokan** [comment]
@janseris, For the Shell page to function properly, at least one Shell content's IsVisible property should be set to true.
In the shared sample, the first Shell content's IsVisible property is set to true, while the remaining Shell contents are set to false through binding. If the first Shell content's IsVisible property is changed to false, all Shell contents would then have their visibility set to false, causing the app to crash. It seems like this is not an issue.

**Comment by LanderVanhove** [comment]
Thanks alot! Using the FlyOutIsVisible instead of IsVisible on the ShellContent is what made it work for me!

Shell:
`    <ShellContent x:Name="ShellLogin" Route="LoginPage" ContentTemplate="{DataTemplate views:LoginPage }"/>

    <FlyoutItem x:Name="BudgeFlyOut" FlyoutDisplayOptions="AsMultipleItems" FlyoutItemIsVisible="False">
        <ShellContent Title="Overview" Route="TransactionsOverview" ContentTemplate="{DataTemplate views:TransactionsOverview }" Icon="overview.png"/>
        <ShellContent Title="New Transaction" Route="SingleTransaction" ContentTemplate="{DataTemplate views:NewTransaction }" Icon="add.png"/>
        <ShellContent Title="Settings" Route="SettingsPage" ContentTemplate="{DataTemplate views:SettingsPage }" Icon="setting.png"/>
        <ShellContent Title="Filters" Route="FilterPage" ContentTemplate="{DataTemplate views:FilterPage }" Icon="filter.png"/>
    </FlyoutItem>`

AppShell.xaml.cs:

**Comment by LanderVanhove** [comment]
Thanks alot! Using the FlyOutIsVisible instead of IsVisible on the ShellContent is what made it work for me! Code now looks like this.

AppShell.xaml:
`<Shell
    x:Class="Budge.AppShell"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:local="clr-namespace:Budge"
    xmlns:views="clr-namespace:Budge.Views"
    Shell.FlyoutBehavior="Flyout"
    Title="Budge"
    Shell.BackgroundColor="{DynamicResource BudgeBackgroundBrown}"
    Style="{StaticResource MyOwnShell}">



    <ShellContent x:Name="ShellLogin" Route="LoginPage" ContentTemplate="{DataTemplate views:LoginPage }"/>

    <FlyoutItem x:Name="BudgeFlyOut" FlyoutDisplayOptions="AsMultipleItems" FlyoutItemIsVisible="False">
        <ShellContent Title="Overview" Route="TransactionsOverview" ContentTemplate="{DataTemplate views:TransactionsOverview }" Icon="overview.png"/>
        <ShellContent Title="New Transaction" Route="SingleTransaction" ContentTemplate="{DataTemplate views:NewTransaction }" Icon="add.png"/>
        <ShellContent Title="Settings" Route="SettingsPage" ContentTemplate="{DataTemplate views:SettingsPage }" Icon="setting.png"/>
        <ShellContent Title="Filters" Route="FilterPage" ContentTemplate="{DataTemplate views:FilterPage }" Icon="filter.png"/>
    </FlyoutItem>
    

</Shell>`

AppShell.xaml.cs:

`        public async Task CheckLoggedIn()
        {
            bool userLoggedIn = await TokenHelper.IsUserLoggedin();

            if (userLoggedIn)
            {
                ShellLogin.IsVisible = false;
                BudgeFlyOut.FlyoutItemIsVisible = true;
            }
            else
            {
                ShellLogin.IsVisible = true;
                BudgeFlyOut.FlyoutItemIsVisible = false;
            }
        }`

**Comment by kubaflo** [comment]
Closing as a usage/support thread. The discussion converged on using FlyoutItemIsVisible rather than toggling ShellContent.IsVisible for this scenario. If there is still a current MAUI bug, please open a new issue with a minimal repro.

**Comment by janseris** [comment]
The issue was that IsVisible caused crash, that seems like a pretty big deal.

**Comment by kubaflo** [comment]
Agreed: a verified crash caused by setting the public `ShellContent.IsVisible` property should not have been treated only as a usage question. `FlyoutItemIsVisible` is a workaround, not proof that the crash is expected. I have reopened this for renewed triage.

> Disclosure: this comment was generated by AI.
